### PR TITLE
[AIRFLOW-2568] Azure Container Instances operator

### PR DESCRIPTION
**Pull request by NielsZeilemaker**
Make sure you have checked _all_ steps below.

### JIRA
- [X] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2568
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [X] Here are some details about my PR, including screenshots of any UI changes:
An Azure Container Instances operator which allows you to run/monitor containers on ACI

### Tests
- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [X] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [X] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [X] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by NielsZeilemaker** [comment]
This is a WIP, still needs documentation etc. Would like to get some comments on the overall structure before I'll add the documentation.

**Comment by kaxil** [comment]
Well, overall structure looks ok to me but I haven't used Azure a lot so wouldn't be able to review that.

File: airflow/contrib/operators/azure_container_instances_operator.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                                                 VolumeMount,
+                                                 ResourceRequests,
+                                                 ResourceRequirements,
+                                                 Container,
+                                                 ContainerGroup)
+from msrestazure.azure_exceptions import CloudError
+
+
+class AzureContainerInstancesOperator(BaseOperator):
+
```
Comment (kaxil) [inline airflow/contrib/operators/azure_container_instances_operator.py:88]: Can you please add a description and a sample usage?

**Review by kaxil — CHANGES_REQUESTED** [review]

**Comment by Fokko** [comment]
Flaky test, can you rebase onto master?

**Pull request closed without merging**